snort/files/snort2pd.py

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env python
+
+from copy import deepcopy
+import datetime
+import hashlib
+import json
+import logging
+from platform import node
+import re
+from subprocess import Popen, PIPE
+import sys
+
+from pygerduty import PagerDuty
+
+
+START_TIME = datetime.datetime.utcnow()
+
+# Log example:
+# 01/09-20:42:02.184887  [**] [1:20000:0] ICMP Packet found [**] [Priority: 0] {ICMP} 10.128.0.3 -> 10.128.0.2
+LOG_FORMAT = r"""
+             (?P<timestamp>\d+/\d+-\d+:\d+:\d+\.\d+)\s+
+             \[\*\*\]\s+
+             \[(?P<gid>\d+):(?P<sid>\d+):(?P<rev>\d+)\]\s+
+             (?P<message>[\w\s!@#$%\^&\*()+=;\'",.?~`/]+)\s+
+             \[\*\*\]\s+
+             \[Priority:\s+(?P<priority>\d+)\]\s+
+             \{(?P<protocol>\w+)\}\s+
+             (?P<src>\d+\.\d+\.\d+\.\d+(:\d+)?)\s+
+             ->\s+
+             (?P<dest>\d+\.\d+\.\d+\.\d+(:\d+)?)
+             """
+
+DEFAULT_CONFIG = {
+    'snort_log_path': '/var/log/snort/alert',
+    'snort_log_format': LOG_FORMAT,
+    'snort_alert_priority': 2,
+    'pagerduty_subdomain': 'example',
+    'pagerduty_api_token': '',
+    'pagerduty_service_key': '',
+}
+
+
+def read_logs(log_path):
+    """Tail a log file and yield lines from it"""
+
+    p = Popen(['tail', '-F', log_path], stdout=PIPE)
```

Comment:
There are a couple of issues with shelling out to  "tail -f" like this. For one I don't think it's feasible to run as a daemon with systemd. What we should do is subscribe to inotify changes on a given file (or folder). An added benefit of this is that we can capture log rotations and update the watch (this implementation is oblivious to log rotation). Please look at this python package: https://pythonhosted.org/watchdog/quickstart.html